utils/get_website_information.py

```diff
@@ -0,0 +1,217 @@
+    links = set()
+    base_domain = urlparse(base_url).netloc
+    for a in soup.find_all("a", href=True):
+        href = a["href"]
+        abs_url = urljoin(base_url, href)
+        if urlparse(abs_url).netloc == base_domain and abs_url.startswith("http"):
+            links.add(abs_url)
+        if len(links) >= max_links:
+            break
+    return list(links)
+
+def extract_seo_info(html):
+    """Extract SEO-relevant info from HTML."""
+    soup = BeautifulSoup(html, "html.parser")
+    seo = {}
+    seo['title'] = soup.title.string.strip() if soup.title and soup.title.string else ''
+    seo['meta_description'] = ''
+    seo['meta_robots'] = ''
+    seo['canonical'] = ''
+    seo['h1'] = []
+    for tag in soup.find_all('meta'):
+        if tag.get('name', '').lower() == 'description':
+            seo['meta_description'] = tag.get('content', '').strip()
+        if tag.get('name', '').lower() == 'robots':
+            seo['meta_robots'] = tag.get('content', '').strip()
+    link_canonical = soup.find('link', rel='canonical')
+    if link_canonical:
+        seo['canonical'] = link_canonical.get('href', '').strip()
+    seo['h1'] = [h1.get_text(strip=True) for h1 in soup.find_all('h1')]
+    return seo
+
+def fetch_robots_txt(base_url):
+    """Fetch robots.txt content for a given base URL."""
+    from urllib.parse import urlparse
+    import requests
+    parsed = urlparse(base_url)
+    robots_url = f"{parsed.scheme}://{parsed.netloc}/robots.txt"
+    try:
+        resp = requests.get(robots_url, timeout=10)
+        if resp.status_code == 200:
+            return resp.text.strip()
+        else:
+            return f"robots.txt not found (status {resp.status_code})"
+    except Exception as e:
+        return f"Error fetching robots.txt: {e}"
+
+async def crawl_and_capture_screenshots(start_url, out_dir, max_pages=5):
+    """Crawl up to max_pages internal pages and capture screenshots and SEO info."""
+    screenshots = []
+    seo_infos = []
+    visited = set()
+    to_visit = [start_url]
+    user_agent = "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36"
+    domain_name = urlparse(start_url).netloc.replace('.', '_')
+    robots_txt = fetch_robots_txt(start_url)
+    # Remove urlparse, use uuid for unique screenshot names
+    async with playwright_browser() as browser:
+        try:
+            context = await browser.new_context(user_agent=user_agent)
+            while to_visit and len(screenshots) < max_pages:
+                url = to_visit.pop(0)
+                if url in visited:
+                    continue
+                visited.add(url)
+                try:
+                    page = await context.new_page()
+                    await page.goto(url, wait_until="load", timeout=30000)
+                    img_path = f"/workspace/{uuid.uuid4().hex[:10]}_screenshot.png"
+                    await page.screenshot(path=img_path, full_page=True)
```

Comment:
Quick question: For single-page applications or long web pages (like https://www.clay.com/), taking a screenshot might miss content such as SOC 2 compliance details, especially if it's further down the page or loaded lazily during scroll.

Would it make sense to include scrolling and wait for the network to be idle (instead of wait till load) to ensure we capture all relevant content? I realize this could add some delay, but I’m thinking about how confident we can be that everything is fully loaded before we take the screenshot.